client/src/AppBundle/Entity/User.php

```diff
@@ -1202,4 +1208,26 @@ public function belongsToActiveOrganisation(): bool
 
         return false;
     }
+
+    public function isAdminOrSuperAdmin(): bool
+    {
+        return $this->getRoleName() === 'ROLE_ADMIN' || $this->getRoleName() === 'ROLE_SUPER_ADMIN';
+    }
+
+    public function canDeleteUser(User $userToBeDeleted): bool
+    {
+        if (!$this->isAdminOrSuperAdmin()) {
+            return false;
+        }
+
+        if ($userToBeDeleted->isAdminOrSuperAdmin()) {
+            return $this->getRoleName() === 'ROLE_SUPER_ADMIN' && $userToBeDeleted->getId() !== $this->getId() ? true : false;
+        }
+
+        if ($userToBeDeleted->getRoleName() === 'ROLE_LAY_DEPUTY' && count($userToBeDeleted->getClients()) <= 1 && $userToBeDeleted->hasReports() === false) {
+            return true;
+        }
+
+        return false;
+    }
```

Comment:
I think this should be part of `UserVoter->determineDeletePermission`, rather than a helper on the model.

In twig, you can then just do `is_granted('delete', user)`.

Reply:
Nice - I had a little voice at the back of my head while I was writing the tests that this feels like a voter now but ignored it as it was the end of the day 😂